temporal-testing-junit5/src/main/java/io/temporal/testing/TestActivityExtension.java

```diff
@@ -95,6 +102,10 @@ public boolean supportsParameter(
       return false;
     }
 
+    if (includesDynamicActivity) {
+      return true;
```

Comment:
Can we add a short javadoc here explaining the logic behind this branch?

Reply:
I was going to write that "No way to check what can be injected if a dynamic activity is registered" but I've realized we can still check that the injected type is a proper activity interface, so I've rewritten the code for that.